pkg/mover/chart.go

```diff
@@ -182,23 +206,37 @@ func NewChartMover(req *ChartMoveRequest, opts ...Option) (*ChartMover, error) {
 		}
 	}
 
-	patternsRaw, err := loadPatterns(req.Source.ImageHintsFile, chart, cm.logger)
+	var hintsRaw []byte
+	if req.Source.Chart.IntermediateBundle != nil {
+		if req.Source.ImageHintsFile != "" {
+			return nil, fmt.Errorf("intermediate bundles already embed the hints file" +
+				", skip explicit hints, they will be ignored")
+		}
+		hintsRaw, err = cm.bundle.loadHints(cm.logger)
+	} else {
+		hintsRaw, err = loadHints(req.Source.ImageHintsFile, cm.chart, cm.logger)
+	}
 	if err != nil {
 		return nil, err
 	}
 
-	if patternsRaw == nil {
+	if hintsRaw == nil {
 		return nil, ErrImageHintsMissing
 	}
 
-	imagePatterns, err := internal.ParseImagePatterns(patternsRaw)
+	imagePatterns, err := internal.ParseImagePatterns(hintsRaw)
 	if err != nil {
 		return nil, fmt.Errorf("failed to parse image patterns: %w", err)
 	}
 
 	cm.logger.Println("Computing relocation...\n")
 
-	imageChanges, err := cm.pullOriginalImages(imagePatterns)
+	var imageChanges []*internal.ImageChange
+	if req.Source.Chart.IntermediateBundle == nil {
+		imageChanges, err = cm.pullOriginalImages(imagePatterns)
+	} else {
+		imageChanges, err = cm.loadImages(cm.bundle, imagePatterns)
+	}
```

Comment:
I am going to simplify this, I only need to call one function, but passing a different loader func.